controllers/metric_controller.go

```diff
@@ -208,49 +208,131 @@ func (r *MetricReconciler) collectMetrics(ctx context.Context, t *redskyv1beta1.
 	return controller.RequeueConflict(err)
 }
 
-func (r *MetricReconciler) target(ctx context.Context, namespace string, m *redskyv1beta1.Metric) (runtime.Object, error) {
-	switch m.Type {
-	case redskyv1beta1.MetricPods:
-		// Use the selector to get a list of pods
-		target := &corev1.PodList{}
-		sel, err := meta.MatchingSelector(m.Selector)
-		if err != nil {
-			return nil, err
-		}
-
-		if err = r.List(ctx, target, client.InNamespace(namespace), sel); err != nil {
-			return nil, err
-		}
+// collectionAttempt updates the status of the trial based on the outcome of an attempt to collect metric values.
+func (r *MetricReconciler) collectionAttempt(ctx context.Context, log logr.Logger, t *redskyv1beta1.Trial, v *redskyv1beta1.Value, probeTime *metav1.Time, err error) (*ctrl.Result, error) {
+	// Do not count retries against the remaining attempts
+	if merr, ok := err.(*metric.CaptureError); ok && merr.RetryAfter > 0 {
+		return &ctrl.Result{RequeueAfter: merr.RetryAfter}, nil
+	}
 
-		return target, nil
-	case redskyv1beta1.MetricPrometheus, redskyv1beta1.MetricJSONPath:
-		// Both Prometheus and JSONPath target a service
-		target := &corev1.ServiceList{}
+	// Update the probe time and ensure that trial observed is still explicitly false (i.e. we have started observation but it is not complete)
+	trial.ApplyCondition(&t.Status, redskyv1beta1.TrialObserved, corev1.ConditionFalse, "", "", probeTime)
 
-		if m.URL != "" {
-			return target, nil
-		}
+	// Decrement the attempts remaining, if there are no attempts left, fail the trial
+	if v.AttemptsRemaining > 0 {
+		v.AttemptsRemaining--
+		if v.AttemptsRemaining == 0 {
+			trial.ApplyCondition(&t.Status, redskyv1beta1.TrialFailed, corev1.ConditionTrue, "MetricFailed", err.Error(), probeTime)
```

Comment:
nit, I think this would still be the same but would save a level
```
if v.AttemptsRemaining > 0 {
  v.AttemptsRemaining--
}
if v.AttemptsRemaining == 0 {
  metric failed.. 
}
```

Reply:
Attempts remaining will be zero coming in on a successful collection attempt, it's only a failure if you decrease to zero.

Reply:
...you could flip it and check for 1:
```
if v.AttemptsRemaining == 1 {
  metric failed.. 
}
if v.AttemptsRemaining > 0 {
  v.AttemptsRemaining--
}
```